procurement/filters.py

```diff
@@ -64,6 +85,20 @@ def filter_postcode(self, queryset, name, value):
 
         return queryset
 
+    def filter_region(self, queryset, name, value):
+        countries = [x[0] for x in Council.COUNTRY_CHOICES]
+        # TODO: Wales + Scotland
+        if value in countries:
+            return queryset.filter(**{"council__nation": value})
+        else:
+            return queryset.filter(**{"council__region": value})
+
+    def filter_notification_frequency(self, queryset, name, value):
+        today = datetime.date.today()
+        difference = datetime.timedelta(days=value)
+        new_value = today - difference
+
+        return queryset.filter(**{"published__gte": new_value})
```

Comment:
I'm not totally sure this is doing the right thing. I think it should be getting contracts that are due to hit one of the 3 month/6 month etc thresholds in the past n days rather than published in the last n days. However, that's just based on reading the page and there's no spec so I could also be wrong.

One other small thing is `earliest_date` or something like that might be a more helpful variable name than `new_value` :)

Reply:
You're absolutely correct! The code (and relevant tests) have now been changed to reflect this :)